src/BillingAddressForm.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useEffect, useState } from "react";
+import Grid from "@mui/material/Unstable_Grid2/Grid2";
+import { Typography, TextField, Box } from "@mui/material";
+
+interface Props {
+  onFormError: () => void;
+  onFormErrorResolved: () => void;
+}
+
+export const BillingAddressForm = ({
+  onFormError,
+  onFormErrorResolved,
+}: Props) => {
+  const [firstNameError, setFirstNameError] = useState(false);
+  const [lastNameError, setLastNameError] = useState(false);
+  const [address1Error, setAddress1Error] = useState(false);
+  const [cityError, setCityError] = useState(false);
+  const [zipError, setZipError] = useState(false);
+  const [countryError, setCountryError] = useState(false);
+  const [stateError, setStateError] = useState(false);
+
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [address1, setAddress1] = useState("");
+  const [address2, setAddress2] = useState("");
+  const [city, setCity] = useState("");
+  const [zip, setZip] = useState("");
+  const [country, setCountry] = useState("");
+  const [state, setState] = useState("");
+
+  useEffect(() => {
+    checkFormErrors();
+
+    return () => {
+      onFormErrorResolved();
+    };
+  }, []);
+
+  useEffect(() => {
+    checkFormErrors();
+  }, [firstName, lastName, address1, city, zip, country, state]);
+
+  const hasErrors =
+    firstNameError ||
+    lastNameError ||
+    address1Error ||
+    cityError ||
+    zipError ||
+    countryError ||
+    stateError;
+
+  const hasEmptyRequiredFields = () => {
```

Comment:
```suggestion
  const hasMissingRequiredFields = () => {
```
nit: This could make it easier to understand the importance of this function.